### PR TITLE
fix: cascader's onChange ts adapts to single and multiple

### PR DESCRIPTION
**Pull request by babycannotsay**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
#33153 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | cascader's multiple determines the declaration of onChange |
| 🇨🇳 Chinese | cascader组件的multiple属性决定onChange的声明类型 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/cascader/index.tsx [hunk 1]
```diff
@@ -80,8 +81,16 @@ const defaultSearchRender: ShowSearchType['render'] = (inputValue, path, prefixC
   return optionList;
 };
 
-export interface CascaderProps<DataNodeType>
-  extends Omit<RcCascaderProps, 'checkable' | 'options'> {
+type SingleCascaderProps = Omit<RcSingleCascaderProps, 'checkable' | 'options'> & {
```
Comment (babycannotsay) [inline components/cascader/index.tsx:84]: 必须在用联合类型前用Omit，否则同一key的声明会被Union
type P = { x: string, z: string } | { x: number }
type C = Omit<P, 'z'> // { x: string | number }

Comment (babycannotsay) [inline components/cascader/index.tsx:84]: 必须在用联合类型前用Omit，否则同一key的声明会被Union
type P = { x: string, z: string } | { x: number }
type C = Omit<P, 'z'> // { x: string | number }

**Comment by babycannotsay** [comment]
#33188 因为files changed打不开，关了重新用这个pr打开了。

**Comment by afc163** [comment]
Please add test case and check failed ci.

**Comment by xjq7** [comment]
image
现在一直报这个错误，onChange类型不兼容

**Comment by Dreaster2** [comment]
image

**Comment by Dreaster2** [comment]
image
Has the problem solved?

**Comment by JianWang-Mc** [comment]
> image
> 
> I install the latest version of antd. It still shows the above problem. Has the problem solved?

我和你一样遇到了这个问题

**Pull request merged**